tests/components/conversation/test_default_agent.py

```diff
@@ -230,6 +234,31 @@ async def test_unexposed_entities_skipped(
     assert result.response.matched_states[0].entity_id == exposed_light.entity_id
 
 
+async def test_state_translations(
+    hass: HomeAssistant,
+    init_components,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test setting state attributes for translated/untranslated state."""
+    exposed_light = entity_registry.async_get_or_create("light", "demo", "1234")
+    hass.states.async_set(
+        exposed_light.entity_id, "off", attributes={ATTR_FRIENDLY_NAME: "demo light"}
+    )
+
+    with patch(
+        "homeassistant.components.conversation.default_agent.template.Template.async_render"
+    ) as render_response:
+        result = await conversation.async_converse(
+            hass, "is the demo light off", None, Context()
+        )
+        assert result.response.response_type == intent.IntentResponseType.QUERY_ANSWER
+        state_attributes = (
+            render_response.call_args[0][0].get("query").get("matched")[0].attributes
+        )
+        assert state_attributes.get(ATTR_TRANSLATED_STATE) == "Off"
```

Comment:
Could we patch translations here so we can get a different value for the translated state?

Reply:
Good point!